aspnetcore/signalr/streaming.md

```diff
@@ -13,35 +13,85 @@ uid: signalr/streaming
+
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.2"
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/live/aspnetcore/signalr/streaming/sample.netcoreapp2.2) ([how to download](xref:index#how-to-download-a-sample))
+
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
 
-## Set up the hub
+[View or download sample code](https://github.com/aspnet/Docs/tree/live/aspnetcore/signalr/streaming/sample.netcoreapp3.0) ([how to download](xref:index#how-to-download-a-sample))
+
+::: moniker-end
 
-A hub method automatically becomes a streaming hub method when it returns a `ChannelReader<T>` or a `Task<ChannelReader<T>>`. Below is a sample that shows the basics of streaming data to the client. Whenever an object is written to the `ChannelReader` that object is immediately sent to the client. At the end, the `ChannelReader` is completed to tell the client the stream is closed.
+## Set up the hub for streaming
+
+### Server to client streaming
+
+A hub method automatically becomes a server to client streaming hub method when it returns a `ChannelReader<T>` or a `Task<ChannelReader<T>>`. Below is a sample that shows the basics of streaming data to the client. Whenever an object is written to the `ChannelWriter` that object is immediately sent to the client. At the end, the `ChannelWriter` is completed to tell the client the stream is closed.
 
 > [!NOTE]
-> * Write to the `ChannelReader` on a background thread and return the `ChannelReader` as soon as possible. Other hub invocations will be blocked until a `ChannelReader` is returned.
+> * Write to the `ChannelWriter` on a background thread and return the `ChannelReader` as soon as possible. Other hub invocations will be blocked until a `ChannelReader` is returned.
 > * Wrap your logic in a `try ... catch` and complete the `Channel` in the catch and outside the catch to make sure the hub method invocation is completed properly.
 
 ::: moniker range="= aspnetcore-2.1"
 
-[!code-csharp[Streaming hub method](streaming/sample/Hubs/StreamHub.aspnetcore21.cs?name=snippet1)]
+[!code-csharp[Streaming hub method](streaming/sample.netcoreapp2.1/Hubs/StreamHub.cs?name=snippet1)]
+
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.2"
+
+[!code-csharp[Streaming hub method](streaming/sample.netcoreapp2.2/Hubs/StreamHub.cs?name=snippet1)]
+
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
+
+[!code-csharp[Streaming hub method](streaming/sample.netcoreapp3.0/Hubs/StreamHub.cs?name=snippet1)]
 
 ::: moniker-end
 
 ::: moniker range=">= aspnetcore-2.2"
 
-[!code-csharp[Streaming hub method](streaming/sample/Hubs/StreamHub.cs?name=snippet1)]
+In ASP.NET Core 2.2 or later, server to client streaming Hub methods can accept a `CancellationToken` parameter that will be triggered when the client unsubscribes from the stream. Use this token to stop the server operation and release any resources if the client disconnects before the end of the stream.
```

Comment:
I'll regret asking this but why not. Adding @scottaddie here to get his 411 on it. Should we say "server-to-client" and "client-to-server" instead of the non-hyphenated versions?

Reply:
👍 for the hyphenation. In this case we have "server to client streaming Hub methods" which really is three adjectives describing the "methods":

1. server-to-client
2. streaming
3. hub

The hyphenation makes this more clear.

Also nit: It looks like in the rest of the docs we don't treat "hub" as a proper noun, so "Hub" in "Hub method" should be lowercase.